Dev/kk/book 19 add logging interceptor

Sadymix opened the pull request.

Commit:
logging request and response data

Commit:
modified configuration of RestTemplate bean for logging purpose

Commit:
removed white space

Commit:
Merge remote-tracking branch 'origin/master' into dev/kk/BOOK-19-add-logging-interceptor

Commit:
removed redundant

Commit:
modified the bean implementation

Commit:
changed incorrect field name refreshType to refreshToken
implemented No args constructor and all args constructor

Commit:
implemented id field for UserDto

Commit:
added 2 records for request and response data
modified intercept method
removed logResponse and logRequest methods and implemented instead setup methods for LogRequestData and LogResponseData

Commit:
changed method name to getRefreshType

Commit:
implemented test cases for LoggingInterceptor

Commit:
fixed the issue with invalid test case implemented method for getting value map

pszerszen commented on src/main/java/com/pgs/client/dto/UserDto.java:
```diff
@@ -12,6 +12,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 public class UserDto {
+    private Integer id;
```
it's safer to use Long for ids

pszerszen commented on src/main/java/com/pgs/client/interceptor/LoggingInterceptor.java:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    private record LogRequestData(URI url, HttpMethod method, HttpHeaders requestHeaders, String requestBody) {
+    }
+
+    private record LogResponseData(HttpStatus statusCode, HttpHeaders headers, String body) {
+    }
```
inner types should be either at the top or bottom of a class, not between methods

pszerszen commented on src/test/java/com/pgs/client/interceptor/LoggingInterceptorTest.java:
```diff
@@ -0,0 +1,101 @@
+    @InjectMocks
+    private LoggingInterceptor loggingInterceptor;
+
+    private static byte[] ARR = new byte[20];
```
![image](https://user-images.githubusercontent.com/15009199/140946155-6ea47d6f-7c13-4e24-864a-da4abc9bb629.png)
give it a better name 😉

pszerszen commented on src/test/java/com/pgs/client/interceptor/LoggingInterceptorTest.java:
```diff
@@ -0,0 +1,101 @@
+
+    @SneakyThrows
+    @Test
+    void testIntercept_badRequest(CapturedOutput capturedOutput) {
```
it's not as much bad request as failed request

pszerszen reviewed (changes requested):
Nice. Just some cosmetic comments this time. Keep it up!

Commit:
changed the id field to wrapper type Long

Commit:
pom

Commit:
moved records to the top of the class

Commit:
changed the byte array name to BODY
changed method name from testIntercept_badRequest to testIntercept_failedRequest

pszerszen commented on src/test/java/com/pgs/client/interceptor/LoggingInterceptorTest.java:
```diff
@@ -0,0 +1,101 @@
+    private ClientHttpRequestExecution clientHttpRequestExecution;
+    @Mock
+    private ClientHttpResponse clientHttpResponse;
+    @InjectMocks
```
nothing to inject
```suggestion

```

Commit:
removed @InjectMocks and initialize loggingInterceptor field
